keadm/cmd/keadm/app/cmd/common/types.go

```diff
@@ -17,37 +17,67 @@ limitations under the License.
 package common
 
 import (
+	"fmt"
+	"strings"
+
 	"github.com/blang/semver"
 )
 
-// InitBaseOptions has the kubeedge cloud deprecated init base information filled by CLI
-type InitBaseOptions struct {
-	KubeEdgeVersion  string
+// CloudInitUpdateBase defines common flags for init and upgrade in the cloud.
+type CloudInitUpdateBase struct {
 	KubeConfig       string
-	Master           string
+	KubeEdgeVersion  string
 	AdvertiseAddress string
-	DNS              string
-	TarballPath      string
+	Profile          string
+	Sets             []string
+	Force            bool
+	DryRun           bool
+}
+
+// GetValidSets returns a valid sets, if the item is an invalid key-value,
+// it is removed from the sets and print the error message.
+func (b CloudInitUpdateBase) GetValidSets() []string {
+	if b.Sets == nil {
+		return nil
+	}
+	res := make([]string, 0, len(b.Sets))
+	for _, s := range b.Sets {
+		p := strings.Split(s, "=")
+		if len(p) < 2 {
+			fmt.Println("Unsupported sets flag: ", s)
+			continue
+		}
+		res = append(res, s)
+	}
+	return res
+}
+
+// HasSets returns the key is in the sets
+func (b CloudInitUpdateBase) HasSets(key string) bool {
+	for _, kv := range b.Sets {
+		if strings.HasPrefix(kv, key) {
```

Comment:
`hasPrefix` may cause incorrect matching. For instance, suppose the existing key is "edgeNode" and the key you want to obtain is edge.